cmd/tempo-vulture/validation.go

```diff
@@ -0,0 +1,286 @@
+
+		err := info.EmitBatches(writer)
+		if err != nil {
+			vs.logger.Error("Failed to write trace", zap.Int("cycle", i+1), zap.Error(err))
+			return traces, err // Any write failure is critical
+		}
+
+		traces = append(traces, traceInfo{
+			id:        info.HexID(),
+			timestamp: timestamp,
+		})
+
+		vs.logger.Info("Wrote trace", zap.Int("cycle", i+1))
+	}
+
+	return traces, nil
+}
+
+func (vs *ValidationService) validateTraceRetrieval(
+	_ context.Context,
+	traces []traceInfo,
+	httpClient httpclient.TempoHTTPClient,
+	result *ValidationResult,
+) {
+	for i, trace := range traces {
+		cycle := i + 1
+		vs.logger.Info("Validating trace retrieval", zap.Int("cycle", cycle), zap.String("traceID", trace.id))
+
+		start := trace.timestamp.Add(-10 * time.Minute).Unix()
+		end := trace.timestamp.Add(10 * time.Minute).Unix()
+
+		retrievedTrace, err := httpClient.QueryTraceWithRange(trace.id, start, end)
+		if err != nil {
+			vs.logger.Error("Failed to read trace", zap.Int("cycle", cycle), zap.Error(err))
+			// failures++
+			result.Failures = append(result.Failures, ValidationFailure{
+				TraceID:   trace.id,
+				Cycle:     cycle,
+				Phase:     "retrieval",
+				Error:     err,
+				Timestamp: trace.timestamp,
+			})
+			continue
+		}
+
+		if len(retrievedTrace.ResourceSpans) == 0 {
+			vs.logger.Error("Retrieved trace has no spans", zap.Int("cycle", cycle))
+			result.Failures = append(result.Failures, ValidationFailure{
+				TraceID:   trace.id,
+				Cycle:     cycle,
+				Phase:     "retrieval",
+				Error:     errors.New("retrieved trace has no spans"),
+				Timestamp: trace.timestamp,
+			})
+			continue
+		}
+		result.SuccessCount++
+
+		vs.logger.Info("Validated trace", zap.Int("cycle", cycle), zap.Int("resourceSpans", len(retrievedTrace.ResourceSpans)))
+	}
+}
+
+func (vs *ValidationService) validateTraceSearch(
+	_ context.Context,
+	traces []traceInfo,
+	httpClient httpclient.TempoHTTPClient,
+	result *ValidationResult,
+) {
+	vs.logger.Info("Waiting for search indexing to complete...")
+
+	vs.logger.Info("writtenTraces", zap.Int("count", len(traces)))
+	vs.logger.Info("Testing search functionality - looking for each written trace...")
+
+	for i, traceInfo := range traces {
+		cycle := i + 1
+
+		// Create a fresh TraceInfo to get the expected attributes
+		info := util.NewTraceInfoWithMaxLongWrites(traceInfo.timestamp, 0, vs.config.TempoOrgID)
+		expected, err := info.ConstructTraceFromEpoch()
+		if err != nil {
+			logger.Error("Failed to construct expected trace for search", zap.Int("cycle", cycle), zap.Error(err))
+			result.Failures = append(result.Failures, ValidationFailure{
+				TraceID:   traceInfo.id,
+				Cycle:     cycle,
+				Phase:     "search",
+				Error:     err,
+				Timestamp: traceInfo.timestamp,
+			})
+			continue
+		}
+
+		// Get a random attribute from the expected trace (same as original vulture)
+		attr := util.RandomAttrFromTrace(expected)
```

Comment:
Is this sufficient for validation mode?  Consider executing a search for each attribute to confirm results.